src/components/Chat/ChatContent/ChatContent.tsx

```diff
@@ -64,14 +64,16 @@ const ChatContent = () => {
               <NewMessageButton messageIndex={-1} />
             )}
             {messages?.map((message, index) => (
-              <React.Fragment key={index}>
-                <Message
-                  role={message.role}
-                  content={message.content}
-                  messageIndex={index}
-                />
-                {!generating && advancedMode && <NewMessageButton messageIndex={index} />}
-              </React.Fragment>
+              (advancedMode || index !== 0) && (
```

Comment:
Great work! This line also needs to check if the first message is the system message.

Reply:
Thank you! I added a check for the system role.